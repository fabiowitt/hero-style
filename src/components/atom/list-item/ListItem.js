import React from 'react';
import PropTypes from 'prop-types';

import css from './ListItem.module.scss';

const ListItem = ({ children }) => {
  return <li className={css['c-list-item']}>{children}</li>;
};

ListItem.propTypes = {
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

ListItem.defaultProps = {
  children: null
};

export default ListItem;
