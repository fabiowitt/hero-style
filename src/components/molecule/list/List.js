import React from 'react';

import PropTypes from 'prop-types';

import css from './List.module.scss';

const List = ({ children }) => <ul className={css['c-list']}>{children}</ul>;

List.propTypes = {
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

List.defaultProps = {
  children: null
};

export default List;
