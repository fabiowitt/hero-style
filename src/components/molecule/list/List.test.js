import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import List from './List';
import ListItem from '../../atom/list-item/ListItem';

describe('List', () => {
  it('should match with the snapshot', () => {
    const children = (
      <ListItem>
        <div>Content</div>
      </ListItem>
    );
    const wrapper = shallow(<List>{children}</List>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
