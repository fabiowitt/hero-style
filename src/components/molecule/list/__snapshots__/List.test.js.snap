// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match with the snapshot 1`] = `
<ul
  className="c-list"
>
  <ListItem>
    <div>
      Content
    </div>
  </ListItem>
</ul>
`;
